### PR TITLE
feat(web-analytics): Add guess for initial host filter

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Some users have asked how to make the web analytics dashboard only show their marketing site. 

## Changes
Add a guess for which hosts are likely to be their marketing site. This guess is based on whether the subdomain is www. or the site is on the domain apex.

## How did you test this code?

Manually

### CODE REVIEW

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.ts [hunk 1]
```diff
@@ -734,12 +745,54 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+                // we try to do the smart thing without people needing to do anything
+                const params = new URLSearchParams([
+                    ['key', '$host'],
+                    ['event_name', '$pageview'],
+                    ['event_name', '$pageleave'],
+                    ['event_name', '$autocapture'],
+                ])
+                let propertiesResponse: unknown
+                try {
+                    propertiesResponse = await api.get(`api/event/values/?${params}`)
```
Comment (mariusandra) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.ts:762]: Minor remark: we do have some kind of API composition system under `api.ts`, freeing us from URL constants. Might be worth hooking into it? 🤷 through if you copied this from elsewhere.

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import api from 'lib/api'
 import { RETENTION_FIRST_TIME, STALE_EVENT_SECONDS } from 'lib/constants'
 import { dayjs } from 'lib/dayjs'
 import { isNotNil } from 'lib/utils'
+import * as tldts from 'tldts'
```
Comment (mariusandra) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.ts:8]: This potentially adds about [100KB of JS](https://bundlephobia.com/package/tldts@6.0.22). Might be worth double checking if so... and perhaps just importing `{ getSubdomain }` solves this?

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.ts [hunk 3]
```diff
@@ -734,12 +745,54 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+                }
+                if (!propertiesResponse || !Array.isArray(propertiesResponse) || propertiesResponse.length <= 1) {
+                    return []
+                }
+                const hosts: string[] = propertiesResponse.map((x) => x.name)
+                // at this point we have multiple hosts, pick the ones that are www or apex
+                const wwwOrApex = hosts.filter((host) => {
+                    const subdomain = tldts.getSubdomain(host)
+                    return subdomain === '' || subdomain === 'www'
+                })
```
Comment (mariusandra) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.ts:774]: Perhaps these subdomains on their own hold value? Suggestion that's non blocking: rename this to a `subdomains` loader, and use a selector to derive the filters.

**Review by mariusandra — COMMENTED** [review]
Looks good, and other than double checking the tldts comment, I don't see any blockers.

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.ts (hunk 2 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.ts:8]: I suspect that this doesn't help much, as it'll still need the big list of 3LDs etc. Will try it anyway!

**Review by mariusandra — APPROVED** [review]
Ah, also, nothing really blocking here, so have a ✅... but I'd still check if there's a way we don't include 100+kb of JS that'll mostly be unused. However if it just loads on the web analytics page, not the app generally, then 🤷

**Pull request closed without merging**